yt/visualization/plot_window.py

```
@@ -2014,3 +2015,97 @@ def SlicePlot(ds, normal=None, fields=None, axis=None, *args, **kwargs):
+            x_fig_size = figure_size
+            y_fig_size = figure_size/aspect
+
+        x_axis_size = ax_text_size[0]
+        y_axis_size = ax_text_size[1]
+
+        top_buff_size = top_buff_size
+
+        xbins = np.array([x_axis_size, x_fig_size, x_axis_size])
+        ybins = np.array([y_axis_size, y_fig_size, top_buff_size])
+
+        size = [xbins.sum(), ybins.sum()]
+
+        x_frac_widths = xbins/size[0]
+        y_frac_widths = ybins/size[1]
+
+        # axrect is the rectangle defining the area of the
+        # axis object of the plot.  Its range goes from 0 to 1 in
+        # x and y directions.  The first two values are the x,y
+        # start values of the axis object (lower left corner), and the
+        # second two values are the size of the axis object.  To get
+        # the upper right corner, add the first x,y to the second x,y.
+        axrect = (
+            x_frac_widths[0],
+            y_frac_widths[0],
+            x_frac_widths[1],
+            y_frac_widths[1],
+        )
+
+        super(LinePlot, self).__init__(size, axrect, None, None)
+        self.add_plot(fields, point1, point2, resolution)
+
+    def add_plot(self, fields, point1, point2, resolution):
+        r"""
+        Used to add plots to the figure
+        """
+        if not isinstance(fields, list):
+            fields = [fields]
+        for field in fields:
+            x, y = self.handler.line_plot(field, np.asarray(point1, dtype='float64'),
+                                          np.asarray(point2, dtype='float64'), resolution)
+            self.axes.plot(x, y)
+
+    def show(self):
+        r"""This will send any existing plots to the IPython notebook.
+
+        If yt is being run from within an IPython session, and it is able to
+        determine this, this function will send any existing plots to the
+        notebook for display.
+
+        If yt can't determine if it's inside an IPython session, it will raise
+        YTNotInsideNotebook.
```

You need to raise this exception manually if you actually want to do this.